pkg/kafka/pubsub_test.go

```diff
@@ -158,3 +161,44 @@ func TestNoGroupSubscriber(t *testing.T) {
 		nil,
 	)
 }
+
+func TestPartitionOffsets(t *testing.T) {
+	pub, sub := newPubSub(t, kafka.DefaultMarshaler{}, "")
+	topicName := "topic_" + watermill.NewUUID()
+
+	var messagesToPublish []*message.Message
+
+	for i := 0; i < 20; i++ {
+		id := watermill.NewUUID()
+		messagesToPublish = append(messagesToPublish, message.NewMessage(id, nil))
+	}
+	err := pub.Publish(topicName, messagesToPublish...)
+	require.NoError(t, err, "cannot publish message")
+
+	messages, err := sub.Subscribe(context.Background(), topicName)
+	require.NoError(t, err)
+
+	receivedMessages, all := subscriber.BulkReadWithDeduplication(messages, len(messagesToPublish), time.Second*10)
+	assert.True(t, all)
```

Comment:
probably we can use require here instead of assert - if we not received all messages test will anyway make no sense